.github/workflows/sync-repos.yml

```diff
@@ -0,0 +1,87 @@
+name: 'Sync Repos'
+
+on: 
+  push:
+    branches: [main]
+
+jobs:
+  check_changes_and_update_repos:
+    name: Code Changes
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read 
+      contents: write 
+      pull-requests: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Check if examples folder has changes
```

Comment:
This step might be unnecessary?  I think it's implicitly handled by the case where `id: check_changes`  results in `changed_paths == ''`

Reply:
I can get rid of this. I just thought that it might be quicker if we do a check to see if any code was changed before iterating through each project in the example directory. Seems like we have nearly 20 example projects at the moment in our org? As we scale, get new customers I'm wondering if we'll continue building many more